Userland/Utilities/mkdir.cpp

```diff
@@ -63,26 +62,27 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
             auto& part = parts[idx];
 
             path_builder.append(part);
-            auto path = path_builder.to_deprecated_string();
+            auto path = path_builder.string_view();
 
-            struct stat st;
-            if (stat(path.characters(), &st) < 0) {
-                if (errno != ENOENT) {
-                    perror("stat");
+            auto stat_or_error = Core::System::stat(path);
+            if (stat_or_error.is_error()) {
+                if (stat_or_error.error().code() != ENOENT) {
+                    warnln("mkdir: {}", strerror(stat_or_error.error().code()));
                     has_errors = true;
                     break;
                 }
 
                 bool is_final = (idx == (num_parts - 1));
                 mode_t mode = is_final ? mask.apply(mask_reference_mode) : default_mode;
 
-                if (mkdir(path.characters(), mode) < 0) {
-                    perror("mkdir");
+                auto maybe_error = Core::System::mkdir(path, mode);
+                if (maybe_error.is_error()) {
+                    warnln("mkdir: {}", strerror(maybe_error.error().code()));
                     has_errors = true;
                     break;
                 }
             } else {
-                if (!S_ISDIR(st.st_mode)) {
+                if (!S_ISDIR(stat_or_error.value().st_mode)) {
```

Comment:
`FileSystem::is_directory()` does a stat call and uses `S_ISDIR(st.st_mode)` internally. I'm not sure the better readability is worth the additional `stat()` call.

(We need the `stat()` call above, rather than just calling `is_directory()` to ensure that doing something like: `touch foo && mkdir -p foo/bar` results in an error; this is how `mkdir` behaves on Linux.)

Reply:
From my point of view, it's more like: "is it worth degrading the readability for premature optimization?".

But it's not like I really mind it, you can wait a maintainer if you want